dfastmi/gui/dialog_view.py

```diff
@@ -142,235 +175,324 @@ def create_qt_application(self) -> None:
-        self.slength.setText(self.view_model.slength)
-        layout.addRow(self.view_model.gui_text("length"), self.slength)
+        self._show_impacted_length(layout)
+
+        # Create conditions group
+        self._create_conditions_group_input(layout)
 
         # get the output directory
-        self.output_dir = QtWidgets.QLineEdit(win)
-        self.output_dir.textChanged.connect(self.output_dir_text_changed)
-        layout.addRow(self.view_model.gui_text("outputDir"), self.openFileLayout(win, self.output_dir, "output_dir"))
+        self._creat_output_directory_input(layout)
 
         # plotting
-        make_plots = QtWidgets.QLabel(self.view_model.gui_text("makePlots"), win)
-        self.make_plots_edit = QtWidgets.QCheckBox(win)
-        self.make_plots_edit.setToolTip(self.view_model.gui_text("makePlots_tooltip"))
-        self.make_plots_edit.stateChanged.connect(self.update_plotting)
-        layout.addRow(make_plots, self.make_plots_edit)
-
-        self.save_plots = QtWidgets.QLabel(self.view_model.gui_text("savePlots"), win)
-        self.save_plots.setEnabled(False)
-        self.save_plots_edit = QtWidgets.QCheckBox(win)
-        self.save_plots_edit.setToolTip(self.view_model.gui_text("savePlots_tooltip"))
-        self.save_plots_edit.stateChanged.connect(self.update_plotting)
-        self.save_plots_edit.setEnabled(False)
-        layout.addRow(self.save_plots, self.save_plots_edit)
-
-        self.figure_dir = QtWidgets.QLabel(self.view_model.gui_text("figureDir"), win)
-        self.figure_dir.setEnabled(False)
-        self.figure_dir_edit = QtWidgets.QLineEdit(win)
-        self.figure_dir_edit.setEnabled(False)
-        self.figure_dir_edit.textChanged.connect(self.figure_dir_text_changed)
-        layout.addRow(self.figure_dir, self.openFileLayout(win, self.figure_dir_edit, "figure_dir_edit"))
-        #dialog["figureDirEditFile"].setEnabled(False)
-
-        self.close_plots = QtWidgets.QLabel(self.view_model.gui_text("closePlots"), win)
-        self.close_plots.setEnabled(False)
-        self.close_plots_edit = QtWidgets.QCheckBox(win)
-        self.close_plots_edit.setToolTip(self.view_model.gui_text("closePlots_tooltip"))
-        self.close_plots_edit.setEnabled(False)
-        self.close_plots_edit.stateChanged.connect(self.update_plotting)
-        layout.addRow(self.close_plots, self.close_plots_edit)
-    
-    def output_dir_text_changed(self, text):
-        if text != self.view_model.output_dir:
-            self.view_model.output_dir = text
+        self._create_make_plots_input_checkbox(layout)
+        self._create_save_plots_input_checkbox(layout)
+        self._create_output_figure_plots_directory_input(layout)        
+        self._create_close_plots_input_checkbox(layout)
+        
+        self._layout.addWidget(self._general_widget)
+
+    def _create_close_plots_input_checkbox(self, layout):
+        self._close_plots = QLabel(self._view_model.gui_text("closePlots"), self._win)
```

Comment:
should all these texts be hardcoded in the gui?
Maybe something to discuss for a followup issue?

Reply:
gui_text is from resource file _messages.LANGUAGE.ini_